.github/dependabot.yml

```diff
@@ -0,0 +1,14 @@
+# Dependabot version updates raises a maximum of five pull requests each time it checks dependencies.
+# Note that there is some overlap with Dependabot security updates so some options can effect security updates as well,
+# see https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.
+
+version: 2
+updates:
```

Comment:
I think we should exclude couple of deps from this:
1. CDK - we bump this one when we need new APIs. otherwise we leave range wide. (this might change in the future but this is what we do today).
2. Data packages. (same reason as CDK).

Or perhaps we should consider `increase-if-necessary` strategy from https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#versioning-strategy and try that first.

I think we should try the strategy first. This will be least noisy option.

Reply:
Makes sense. The `increase-if-necessary` strategy would still try to bump CDK and Data packages if there is a new major version so maybe we still want to exclude them?

Reply:
I'd keep them included if we adopt that strategy. We'll get notified about the fact that MV was shipped.

My main concern was to be less aggressive with updates for these (in package.jsons, we have `test_with_baseline_deps` in healthchecks to cover bottom of the range). The `increase-if-necessary` mitigates that concern.

Reply:
Now my only concern is how `increase-if-necessary` plays with security updates as docs don't clearly state what happens (just that security updates are affected).

For example with https://github.com/aws-amplify/amplify-backend/pull/2253, since these were all patch versions would Dependabot even make the PR?

Reply:
Looks like it will still work, I reverted the change in #2253 in my fork and added the versioning strategy. https://github.com/rtpascual/amplify-backend-fork/pull/12

Reply:
This https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/customizing-dependency-updates#impact-of-configuration-changes-on-security-updates 
says `These pull requests are always triggered by a security advisory for a dependency, rather than by the Dependabot schedule. However, they inherit relevant configuration settings from the dependabot.yml file unless you specify a different target branch for version updates.`

So we'll keep getting these PRs as soon as security alert is present.

Previous PRs usually changed package-lock. only which would correspond to `lockfile-only` or `auto` going into `lockfile-only` mode. If I understand this correctly the `increase-if-necessary` is `lockfile-only` for minor-patch bumps, package.json + lock file for mv bumps.

Reply:
Ah I missed that particular section. I was reading this which led me to security update related doc pages where I didn't find anything helpful:
https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file
`In general, security updates use any configuration options that affect pull requests, for example, adding metadata or changing their behavior. For more information about security updates, see [Configuring Dependabot security updates](https://docs.github.com/en/code-security/dependabot/dependabot-security-updates/configuring-dependabot-security-updates).`